app/assets/javascripts/discourse/controllers/login.js.es6

```diff
@@ -114,57 +106,62 @@ export default Ember.Controller.extend(ModalFunctionality, {
       ajax("/session", {
         type: "POST",
         data: {
-          login: this.get("loginName"),
-          password: this.get("loginPassword"),
-          second_factor_token: this.get("secondFactorToken"),
-          second_factor_method: this.get("secondFactorMethod")
+          login: this.loginName,
+          password: this.loginPassword,
+          second_factor_token: this.secondFactorToken,
+          second_factor_method: this.secondFactorMethod
         }
       }).then(
-        function(result) {
+        result => {
           // Successful login
           if (result && result.error) {
-            self.set("loggingIn", false);
+            this.set("loggingIn", false);
             if (
               result.reason === "invalid_second_factor" &&
-              !self.get("secondFactorRequired")
+              !this.secondFactorRequired
             ) {
-              $("#modal-alert").hide();
-              self.setProperties({
+              document.getElementById("modal-alert").style.display = "none";
+
+              this.setProperties({
                 secondFactorRequired: true,
                 showLoginButtons: false,
                 backupEnabled: result.backup_enabled,
                 showSecondFactor: true
               });
 
-              Ember.run.next(() => {
-                $("#second-factor input").focus();
-              });
+              Ember.run.schedule("afterRender", () =>
+                document.getElementById("second-factor input").focus()
```

Comment:
This seems odd to me. Did you want a querySelector here? Because input is not part of the id.

Reply:
yes indeed